src/textual/widgets/_select.py

```diff
@@ -0,0 +1,376 @@
+
+
+class SelectCurrent(Horizontal):
+    """Displays the currently selected option."""
+
+    DEFAULT_CSS = """
+    SelectCurrent {
+        border: tall $background;
+        background: $boost;
+        color: $text;
+        width: 100%;
+        height: auto;
+        padding: 0 2;
+    }
+    SelectCurrent Static#label {
+        width: 1fr;
+        height: auto;
+        color: $text-disabled;
+        background: transparent;
+    }
+    SelectCurrent.-has-value Static#label {
+        color: $text;
+    }
+    SelectCurrent .arrow {
+        box-sizing: content-box;
+        width: 1;
+        height: 1;
+        padding: 0 0 0 1;
+        color: $text-muted;
+        background: transparent;
+    }
+    SelectCurrent .arrow {
+        box-sizing: content-box;
+        width: 1;
+        height: 1;
+        padding: 0 0 0 1;
+        color: $text-muted;
+        background: transparent;
+    }
+    """
+
+    has_value = var(False)
+    """True if there is a current value, or False if it is None."""
+
+    class Toggle(Message):
+        """Request toggle overlay."""
+
+    def __init__(self, placeholder: str) -> None:
+        """Initialize the SelectCurrent.
+
+        Args:
+            placeholder: A string to display when there is nothing selected.
+        """
+        super().__init__()
+        self.placeholder = placeholder
+        self.label: RenderableType | None = None
+
+    def update(self, label: RenderableType | None) -> None:
+        """Update the content in the widget.
+
+        Args:
+            label: A renderable to display, or `None` for the placeholder.
+        """
+        self.label = label
+        self.has_value = label is not None
+        self.query_one("#label", Static).update(
+            self.placeholder if label is None else label
+        )
+
+    def compose(self) -> ComposeResult:
+        """Compose label and down arrow."""
+        yield Static(self.placeholder, id="label")
+        yield Static("▼", classes="arrow down-arrow")
+        yield Static("▲", classes="arrow up-arrow")
+
+    def _watch_has_value(self, has_value: bool) -> None:
+        """Toggle the class."""
+        self.set_class(has_value, "-has-value")
+
+    async def _on_click(self, event: events.Click) -> None:
+        """Inform ancestor we want to toggle."""
+        self.post_message(self.Toggle())
+
+
+SelectType = TypeVar("SelectType")
+"""The type used for data in the Select."""
+SelectOption: TypeAlias = "tuple[str, SelectType]"
+"""The type used for options in the Select."""
+
+
+class Select(Generic[SelectType], Vertical, can_focus=True):
+    """Widget to select from a list of possible options.
+
+    A Select displays the current selection.
+    When activated with ++enter++ the widget displays an overlay with a list of all possible options.
+
+    """
+
+    BINDINGS = [("enter", "show_overlay")]
+    """
+    | Key(s) | Description |
+    | :- | :- |
+    | enter | Activate the overlay |
```

Comment:
<kbd>Space</kbd> is another common keystroke to active such a control (see macOS for example). I think I'd be tempted to support that too, perhaps even down arrow as well.